src/java.base/unix/native/libjli/java_md.c

```diff
@@ -375,12 +375,33 @@ CreateExecutionEnvironment(int *pargc, char ***pargv,
          *     o          $JVMPATH (directory portion only)
          *     o          $JRE/lib
          *     o          $JRE/../lib
-         *     o          ZLIBNX_PATH (for AIX P9 or newer systems with NX)
+         *     o          ZLIBNX_PATH (for AIX P9 or newer systems with NX, unless -XX:-UseZlibNX is set)
          *
          * followed by the user's previous effective LD_LIBRARY_PATH, if
          * any.
          */
 
+#ifdef AIX
+        int argc = *pargc;
+        char **argv = *pargv;
+        char *arg = NULL;
+        jboolean useZlibNX = JNI_TRUE;
+        while (argc > 0 && *(arg = *argv) == '-') {
```

Comment:
This needs to skip the first element of `argv` which is the name of the launcher:
```
Command line args:
argv[0] = ./jdk/bin/java
argv[1] = -XX:-UseZlibNX
argv[2] = -version
```
I suggest:
```c
        int argc = *pargc - 1;
        char **argv = *pargv + 1;
        jboolean useZlibNX = JNI_TRUE;
        for (; argc > 0; --argc, ++argv) {
            const char *arg = *argv;
            if ('-' != *arg) {
                break;
            } else if (JLI_StrCmp(arg, "-XX:+UseZlibNX") == 0) {
...
```

Reply:
I've updated it. The while loop is the same as used in java.c SelectVersion() and ParseArgument(), I think we can keep it as is?